contracts/RRUtils.sol

```diff
@@ -152,4 +152,47 @@ library RRUtils {
 
         return false;
     }
+
+    function compareLabel(bytes memory self, bytes memory other) internal  returns (int){
+        uint sLength = labelCount(self, 0);
+        uint oLength = labelCount(other, 0);
+        uint shortest = sLength;
+
+        if (shortest > oLength) { shortest = oLength; }
+        bytes memory sHead;
+        bytes memory oHead;
+        if(sLength == shortest){ oHead = head(other, 0);}
```

Comment:
Is this necessary? If one is a substring of the other, we only need to know which is shorter, not what the extra label(s) are.